### PR TITLE
Make mapNextTx private within CTxMemPool

### PR DESCRIPTION
**Pull request by TheBlueMatt**
This builds on #5206, removing the (now duplicate) check from AcceptToMemoryPool and makes mapNextTx private in CTxMemPool.

### CODE REVIEW

**Comment by laanwj** [comment]
@morcos has your comment been resolved here?

**Comment by morcos** [comment]
Yes that problem was fixed.  But my [comment](https://github.com/bitcoin/bitcoin/pull/5206#issuecomment-64681917) on the other PR still applies in that this changes behavior.  As far as I could find the only change is in rpcsignrawtransaction and preventing signing of double spends (now also when the inputs are in the chain) and maybe that's ok?   You can see this by switching the order of the `tx2 = ...` and `spend_id = ...` lines in the included rpc test.  Switching them will work prior to this change, but not with it.

File: qa/rpc-tests/mempool_reject_doublespend.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+from util import *
+import os
+import shutil
+
+# Create one-input, one-output, no-fee transaction:
+class MempoolSpendCoinbaseTest(BitcoinTestFramework):
+
+    def setup_network(self):
+        # Just need one node for this test
+        args = ["-checkmempool", "-debug=mempool"]
```
Comment (morcos) [inline qa/rpc-tests/mempool_reject_doublespend.py]: needs "-relaypriority=0" or to create the tx with a fee.

Comment (TheBlueMatt) [inline qa/rpc-tests/mempool_reject_doublespend.py]: Fixed.

**Comment by laanwj** [comment]
Any way forward on this? Or better to close for now?

**Comment by TheBlueMatt** [comment]
@laanwj I'm not sure what your question was? Was this reviewable on Nov 4? Yes, unless people rejected to the change described by @morcos (no one voiced any opinion either way, AFAIR). But, yes, this now needs rebase due to mempool changes. I'll try to get to that next week or so.

**Comment by TheBlueMatt** [comment]
I went ahead and rebased, in case anyone is interested in this, but the API change is maybe not great. Instead of being able to sign double-spends against unconfirmed txn freely, with this change you have to provide the inputs explicitly. Its not so hard to do, but maybe isnt such a nice API change.

**Pull request closed without merging**